tests/TestCase/Routing/Route/RouteTest.php

```diff
@@ -1578,6 +1578,40 @@ public function testUTF8PatternOnSection(): void
         $this->assertEquals($expected, $result);
     }
 
+    public function testUrlEncodedSlash(): void
+    {
+        $route = new Route(
+            '/products/tests/*',
+            ['controller' => 'Products', 'action' => 'test'],
+            ['_urldecode' => false]
+        );
+
+        $result = $route->parse('/products/tests/xx%2Fyy', 'GET');
+        $expected = [
+            'controller' => 'Products',
+            'action' => 'test',
+            'pass' => ['xx/yy'],
```

Comment:
So an unmatched passed parameter will always be decoded once while a matched one will not?

Reply:
That feels off. We could re-urlencode the pass values, or disable the urldecoding there as well.

Reply:
This decoding is due to the `rawurldecode()` in `_parseArgs()` which is called **only** for the part that matches the trailing `*` in the route template. As you can see in next test, no decoding is done for the value matching the named placeholder `{slug}`.

Also as per the issue #15269, he expects the decoded value `xx/yy` and not `xx%2Fyy` in passed args.

So maybe this special casing for `*` is intentional? Or perhaps this decoding should be done for `**` instead of `*`?

Reply:
> I think having the * use the route's urldecode setting instead.

Okay, I will update accordingly.